### PR TITLE
Implement channel query and ChanOpenInit message

### PR DESCRIPTION
**Pull request by ancazamfir**
Closes: #64

## Description
Added the channel query CLI, request/ response structs, channel struct, ChanOpenInit message and tests.
The other messages will be added as part of a different PR.
______

For contributor use:

- [X] Unit tests written
- [ ] Added test to CI if applicable 
- [ ] Updated CHANGELOG_PENDING.md
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: modules/src/ics04_channel/msgs.rs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                p.counterparty_port_id,
+                p.counterparty_channel_id,
+                acc,
+            );
+
+            match msg {
+                Ok(_res) => {
+                    assert!(
+                        test.want_pass,
+                        "MsgConnOpenInit::new should have failed for test {}, \nmsg {:?}",
```
Comment (Shivani912) [inline modules/src/ics04_channel/msgs.rs:200]:
```suggestion
                        "MsgChanOpenInit::new should have failed for test {}, \nmsg {:?}",
```

File: modules/src/ics04_channel/msgs.rs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                        test.name,
+                        test.params.clone()
+                    );
+                }
+                Err(err) => {
+                    assert!(
+                        !test.want_pass,
+                        "MsgConnOpenInit::new failed for test {}, \nmsg {:?} with err {:?}",
```
Comment (Shivani912) [inline modules/src/ics04_channel/msgs.rs:208]:
```suggestion
                        "MsgChanOpenInit::new failed for test {}, \nmsg {:?} with err {:?}",
```

**Comment by ancazamfir** [comment]
> Looks good to me! But some tests are failing. Let's fix that and then merge :)

Not sure how to fix this. Will try, it fails in the client also (on master)

**Comment by ancazamfir** [comment]
> > Looks good to me! But some tests are failing. Let's fix that and then merge :)
> 
> Not sure how to fix this. Will try, it fails in the client also (on master)

I will work on this in a separate PR so I unblock you.

**Pull request merged**